sympy/stats/stochastic_process_types.py

```diff
@@ -326,6 +324,104 @@ def _preprocess(self, given_condition, evaluate):
 
         return is_insufficient, trans_probs, state_space, given_condition
 
+    def _transient2transient(self):
+        """
+        Computes the one step probabilities of transient
+        states to transient states. Used in finding
+        fundamental matrix, absorbing probabilties.
+        """
+        trans_probs = self.transition_probabilities
+        if not isinstance(trans_probs, ImmutableMatrix):
+            return None
+
+        m, trans_states, t2t = trans_probs.shape[0], [], []
+        for i in range(m):
+            if trans_probs[i, i] != 1:
+                trans_states.append(i)
```

Comment:
Or just: `trans_states = [i for i in range(m) if trans_probs[i, i] != 1]`.

Reply:
Thanks. I was also looking for one line alternative.  Will update this.